lib/ts/recipe/emailpassword/recipeImplementation.ts

```diff
@@ -77,28 +90,19 @@ export default function getRecipeInterface(
                 input.userContext
             );
             if (resp.status === "OK") {
-                resp.user = new User(resp.user);
-                resp.recipeUserId = new RecipeUserId(resp.recipeUserId);
+                return {
+                    status: "OK",
+                    user: new User(resp.user),
+                    recipeUserId: new RecipeUserId(resp.recipeUserId),
+                };
             }
             return resp;
 
             // we do not do email verification here cause it's a new user and email password
             // users are always initially unverified.
         },
 
-        signIn: async function ({
-            email,
-            password,
-            tenantId,
-            userContext,
-        }: {
-            email: string;
-            password: string;
-            tenantId: string;
-            userContext: any;
-        }): Promise<
-            { status: "OK"; user: UserType; recipeUserId: RecipeUserId } | { status: "WRONG_CREDENTIALS_ERROR" }
-        > {
+        signIn: async function ({ email, password, tenantId, session, userContext }) {
```

Comment:
sign in should not call core directly. Instead, it should call verifyCredentials

Reply:
`verifyCredentials` has a different type (intentionally), that doesn't include the user, because we do not want people to use this for sign-in.
I can change that, and maybe remove the user object on the index fn level, but that'd be confusing I think.